internal/pkg/api/hardware.go

```diff
@@ -3,32 +3,26 @@ package api
 import (
 	"fmt"
 	"os"
+	"path/filepath"
 
+	"github.com/aws/eks-anywhere/pkg/providers/tinkerbell/hardware"
 	"github.com/gocarina/gocsv"
 )
 
 const (
-	Dell       = "dell"
-	HP         = "hp"
-	SuperMicro = "supermicro"
+	HardwareVendorDell       = "dell"
+	HardwareVendorHP         = "hp"
+	HardwareVendorSuperMicro = "supermicro"
+	HardwareVendorAgnostic   = "agnostic"
 )
 
 type Hardware struct {
-	Id           string `csv:"guid"`
-	IpAddress    string `csv:"ip_address"`
-	Gateway      string `csv:"gateway"`
-	Nameservers  string `csv:"nameservers"`
-	Netmask      string `csv:"netmask"`
-	MacAddress   string `csv:"mac"`
-	Hostname     string `csv:"hostname"`
-	Vendor       string `csv:"vendor"`
-	BmcIpAddress string `csv:"bmc_ip"`
-	BmcUsername  string `csv:"bmc_username"`
-	BmcPassword  string `csv:"bmc_password"`
+	Id string `csv:"guid"`
+	hardware.Machine
 }
 
 func NewHardwareSlice(csvFile string) ([]*Hardware, error) {
```

Comment:
I tend to make it explicit that we're reading from a file when performing initialization from a file (e.g. `NewHardwareSliceFromFile()`). The standard `New` func shouldn't really be doing anything fancy like opening files. It _could_ be given an `io.Reader` though allowing you to deal with opening files as a separate concern from initialization. Then you can do stuff like:

```go
func NewHardwareSlice(r io.Reader) ... {...}

func NewHardwareSliceFromFile(file string) ... {
  fh, _ := os.Open(file, ...)
  return NewHardwareSlice(fh)
}
```